src/component/helpers/array-utils.ts

```diff
@@ -55,10 +55,13 @@ export function ensureIsArray<T>(elements: (T | string)[] | T | string, acceptEm
 export function filter<T extends string>(arrayToFilter: T[], suffix: string, options?: FilterParameter): T[] {
   let pattern = '';
   if (options?.startingWith) {
+    // eslint-disable-next-line unicorn/prefer-spread -- pattern is a string not an array
```

Comment:
I don't understand why the rule detects it as `string.concat` and not `array.concat`. If so, it is a bug in the rule 
I am testing this locally

Reply:
See https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2041